client/protocols/unlock.py

```diff
@@ -153,12 +153,14 @@ def interact(self, unique_id, case_id, question_prompt, answer, choices=None, ra
                 correct = True
             tg_id = -1
             misU_count_dict = {}
+            rationale = "Unknown"
 
             if not correct:
-                misU_count_dict, tg_id, printed_msg = self.guidance_util.show_guidance_msg(unique_id,input_lines,
-                    self.hash_key)
-
+                guidance_data = self.guidance_util.show_guidance_msg(unique_id, input_lines,
+                                                                     self.hash_key)
+                misU_count_dict, tg_id, printed_msg, rationale = guidance_data
             else:
+                rationale = self.guidance_util.prompt_with_prob()
```

Comment:
I'd say it should be very rare. I told Hilfinger it'd be about once per question set. Most question sets have only <10 questions. So how about a probability of 10% with halving it from there?